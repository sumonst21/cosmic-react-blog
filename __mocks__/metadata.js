export default {
  finishedLoadAt: 1526859679840,
  isLoading: false,
  logo: {
    url: 'url.png',
    imgix_url: 'ingix_url.png',
  },
  startedLoadAt: 1526859679433,
  tag: '"Content is king" -- Bill Gates',
  title: 'My Amazing Blog',
};
