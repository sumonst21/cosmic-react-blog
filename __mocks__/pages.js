export default {
  1: {
    finishedLoadAt: 1526859679918,
    isLoading: false,
    startedLoadAt: 1526859679433,
    slugs: ['a-wonderful-blog-post-about-earth', 'another-wonderful-blog-post-about-earth'],
  },
};
