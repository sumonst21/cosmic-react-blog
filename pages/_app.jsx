import React from 'react';
import App, { Container } from 'next/app';
import { Provider } from 'react-redux';
import { PageTransition } from 'next-page-transitions';

import withReduxStore from '../lib/withReduxStore';
import AppContainer from '../components/AppContainer';

class CustomApp extends App {
  componentDidUpdate() {
    if (!window.previouslyLoaded) {
      window.previouslyLoaded = true;
    }
  }

  render() {
    const { Component, pageProps, reduxStore } = this.props;

    return (
      <Container>
        <Provider store={reduxStore}>
          <AppContainer>
            <PageTransition
              timeout={300}
              classNames="page-transition"
            >
              <Component {...pageProps} />
            </PageTransition>
            <style jsx global>
              {`
                .page-transition-enter {
                  opacity: 0;
                }
                .page-transition-enter-active {
                  opacity: 1;
                  transition: opacity 300ms;
                }
                .page-transition-exit {
                  opacity: 1;
                }
                .page-transition-exit-active {
                  opacity: 0;
                  transition: opacity 300ms;
                }
              `}
            </style>
          </AppContainer>
        </Provider>
      </Container>
    );
  }
}

CustomApp.getInitialProps = async ({ Component, ctx }) => {
  let pageProps = {};

  if (Component.getInitialProps) {
    pageProps = await Component.getInitialProps(ctx);
  }

  return { pageProps };
};

export { CustomApp as DisconnectedCustomApp };

export default withReduxStore(CustomApp);
