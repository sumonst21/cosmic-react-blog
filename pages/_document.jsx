/* Adapted from: https://github.com/mui-org/material-ui/blob/master/examples/nextjs/pages/_document.js */
import React, { Fragment } from 'react';
import Document, { Head, Main, NextScript } from 'next/document';
import JssProvider from 'react-jss/lib/JssProvider';
import flush from 'styled-jsx/server';

import getPageContext from '../src/getPageContext';

class CustomDocument extends Document {
  render() {
    const { pageContext } = this.props;

    return (
      <html lang="en" dir="ltr">
        <Head>
          <title>Cosmic Blog</title>
          <meta charSet="utf-8" />
          <meta
            name="viewport"
            content={
              'user-scalable=0, initial-scale=1, ' +
              'minimum-scale=1, width=device-width, height=device-height'
            }
          />
          <link rel="apple-touch-icon" sizes="180x180" href="/static/apple-touch-icon.png" />
          <link rel="icon" type="image/png" sizes="32x32" href="/static/favicon-32x32.png" />
          <link rel="icon" type="image/png" sizes="16x16" href="/static/favicon-16x16.png" />
          <link rel="manifest" href="/static/manifest" />
          <meta name="msapplication-TileColor" content="#da532c" />
          <meta name="theme-color" content={pageContext.theme.palette.primary.main} />
          <link
            href="https://fonts.googleapis.com/css?family=Lora|Montserrat"
            rel="stylesheet"
          />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </html>
    );
  }
}

CustomDocument.getInitialProps = (ctx) => {
  const pageContext = getPageContext();
  const page = ctx.renderPage(Component => props => (
    <JssProvider
      registry={pageContext.sheetsRegistry}
      generateClassName={pageContext.generateClassName}
    >
      <Component pageContext={pageContext} {...props} />
    </JssProvider>
  ));

  return {
    ...page,
    pageContext,
    styles: (
      <Fragment>
        <style
          id="jss-server-side"
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: pageContext.sheetsRegistry.toString() }}
        />
        {flush() || null}
      </Fragment>
    ),
  };
};

export default CustomDocument;
