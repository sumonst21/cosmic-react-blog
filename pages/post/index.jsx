import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import classnames from 'classnames';
import moment from 'moment';
import Head from 'next/head';

import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import withRoot from '../../src/withRoot';
import { fetchLinksIfNeeded } from '../../state/links/actions';
import { fetchPostIfNeeded } from '../../state/posts/actions';
import AppBar from '../../components/AppBar';
import AuthorInfo from '../../components/AuthorInfo';
import SocialLinks from '../../components/SocialLinks';
import { BASE_URL } from '../../utils/constants';

const styles = theme => ({
  authorContainer: {
    alignItems: 'center',
    display: 'flex',
    marginBottom: theme.spacing.unit * 2,
  },
  authorInfo: {
    margin: theme.spacing.unit * 6,
    marginTop: theme.spacing.unit * 4,
  },
  authorFigure: {
    height: 16,
    margin: 0,
    marginRight: theme.spacing.unit,
    width: 16,
  },
  centerColumn: {
    [theme.breakpoints.down('xs')]: {
      gridColumn: '1 / span 1',
    },
    gridColumn: '2 / span 1',
  },
  date: {
    marginBottom: theme.spacing.unit * 2,
  },
  divider: {
    border: 0,
    borderTop: `1px solid ${theme.palette.text.secondary}`,
    height: 0,
    marginBottom: theme.spacing.unit * 2,
    opacity: 0.2,
    width: '100%',
  },
  fullWidth: {
    [theme.breakpoints.down('xs')]: {
      gridColumn: '1 / span 1',
    },
    gridColumn: '1 / span 3',
  },
  heroFigure: {
    display: 'block',
    margin: 0,
    overflow: 'hidden',
    padding: `${theme.spacing.unit * 2}px 0`,
  },
  heroImg: {
    display: 'block',
    margin: 0,
    padding: 0,
    width: '100%',
  },
  main: {
    ...theme.mixins.gutters(),
    [theme.breakpoints.down('xs')]: {
      gridTemplateColumns: 'auto',
    },
    gridTemplateColumns: `${theme.spacing.unit * 4}px auto ${theme.spacing.unit * 4}px`,
    display: 'grid',
    marginBottom: theme.spacing.unit * 4,
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 0,
    maxWidth: 900,
  },
  root: {
    paddingTop: theme.spacing.unit * 6,
    paddingBottom: theme.spacing.unit * 6,
  },
  title: {
    marginBottom: theme.spacing.unit * 2,
  },
});

const Post = ({
  asPath,
  classes,
  content,
  metadata,
  published_at, // eslint-disable-line camelcase
  title,
}) => {
  const author = metadata && metadata.author;
  const authorName = author && author.title;
  const authorImageUrl = author && author.metadata && author.metadata.image
    && author.metadata.image.imgix_url;
  const heroUrl = metadata && metadata.hero && metadata.hero.imgix_url;

  return (
    <Fragment>
      <Head>
        <title>{title}</title>
        {heroUrl && <meta property="og:image" content={`${heroUrl}?w=279&h=279&fit=crop&fm=png`} key="og_image" />}
        {heroUrl && <meta property="og:image:type" content="image/png" key="og_image_type" />}
        {heroUrl && <meta property="og:image:width" content="279" key="og_image_width" />}
        {heroUrl && <meta property="og:image:height" content="279" key="og_image_height" />}
        {content && <meta property="og:description" content={content} key="og_description" />}
        {title && <meta property="og:title" content={title} key="og_title" />}
        {asPath && <meta property="og:url" content={`${BASE_URL}${asPath}`} key="og_url" />}
      </Head>
      <AppBar title={title} />
      <div className={classes.root}>
        <main className={classes.main}>
          {
            (authorName || authorImageUrl) &&
            <div
              className={classnames([
                classes.authorContainer,
                classes.centerColumn,
              ])}
            >
              {
                authorImageUrl &&
                <figure className={classes.authorFigure}>
                  <img
                    alt=""
                    heght="16px"
                    src={`${authorImageUrl}?w=16&h=16&fit=crop`}
                    width="16px"
                  />
                </figure>
              }
              {
                authorName &&
                <Typography variant="display1">
                  {authorName}
                </Typography>
              }
            </div>
          }
          {
            title &&
            <Typography
              classes={{
                root: classnames([
                  classes.centerColumn,
                  classes.title,
                ]),
              }}
              variant="display3"
            >
              {title}
            </Typography>
          }
          {
            published_at && // eslint-disable-line camelcase
            <Typography
              classes={{
                root: classnames([
                  classes.centerColumn,
                  classes.date,
                ]),
              }}
              variant="subheading"
            >
              {moment(published_at).format('MMMM Do, YYYY')}
            </Typography>
          }
          <hr
            className={classnames([
            classes.centerColumn,
            classes.divider,
          ])}
          />
          {
            heroUrl &&
            <figure
              className={classnames([
                classes.heroFigure,
                classes.fullWidth,
              ])}
            >
              <img
                alt=""
                className={classes.heroImg}
                src={`${heroUrl}?w=900&fit=crop`}
              />
            </figure>
          }
          {
            content &&
            <span className={classes.centerColumn}>
              <Typography variant="body1" component="span">
                {/* eslint-disable-next-line react/no-danger */}
                <span dangerouslySetInnerHTML={{ __html: content }} />
              </Typography>
            </span>
          }
          <hr
            className={classnames([
            classes.centerColumn,
            classes.divider,
          ])}
          />
          {
            author &&
            <AuthorInfo
              {...author}
              className={classnames([
                classes.centerColumn,
                classes.authorInfo,
              ])}
            />
          }
          <hr
            className={classnames([
              classes.centerColumn,
              classes.divider,
            ])}
          />
        </main>
        <SocialLinks />
      </div>
    </Fragment>
  );
};

Post.getInitialProps = async ({ query, reduxStore }) => {
  const { slug } = query;
  const { dispatch } = reduxStore;

  try {
    await Promise.all([
      dispatch(fetchPostIfNeeded(slug)),
      dispatch(fetchLinksIfNeeded()),
    ]);
    const { posts } = reduxStore.getState();
    const { post } = posts[slug];
    return { ...post };
  } catch (e) {
    console.error(e); // eslint-disable-line no-console
    return { slug };
  }
};

Post.defaultProps = {
  asPath: undefined,
  content: undefined,
  metadata: undefined,
  published_at: undefined,
  title: undefined,
};

Post.propTypes = {
  asPath: PropTypes.string,
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  content: PropTypes.string,
  metadata: PropTypes.shape({
    author: PropTypes.shape({
      content: PropTypes.string,
      metadata: PropTypes.shape({
        image: PropTypes.shape({
          imgix_url: PropTypes.string,
        }),
      }),
      title: PropTypes.string,
    }),
    hero: PropTypes.shape({
      imgix_url: PropTypes.string,
    }),
  }),
  published_at: PropTypes.string,
  title: PropTypes.string,
};

export const BasicPost = withStyles(styles)(Post);

export default compose(
  withRoot,
  withStyles(styles),
)(Post);
