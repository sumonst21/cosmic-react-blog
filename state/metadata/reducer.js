import { actionTypes } from './actions';

const {
  CATCH_METADATA_FETCH_ERROR,
  RECEIVE_SITE_METADATA,
  REQUEST_SITE_METADATA,
} = actionTypes;

const defaultRootState = {
  error: undefined,
  finishedLoadAt: undefined,
  isLoading: false,
  logo: undefined,
  startedLoadAt: undefined,
  tag: undefined,
  title: undefined,
};

const rootReducer = (state = defaultRootState, action) => {
  switch (action.type) {
    case CATCH_METADATA_FETCH_ERROR:
      return {
        ...state,
        error: action.error,
        finishedLoadAt: Date.now(),
        isLoading: false,
      };
    case RECEIVE_SITE_METADATA:
      return {
        ...state,
        error: undefined,
        finishedLoadAt: Date.now(),
        isLoading: false,
        logo: action.logo,
        tag: action.tag,
        title: action.title,
      };
    case REQUEST_SITE_METADATA:
      return {
        ...state,
        error: undefined,
        isLoading: true,
        startedLoadAt: Date.now(),
      };
    default:
      return state;
  }
};

export default rootReducer;
