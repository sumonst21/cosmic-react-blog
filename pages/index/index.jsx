import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';
import { compose } from 'recompose';
import Head from 'next/head';

import Typography from '@material-ui/core/Typography';

import withRoot from '../../src/withRoot';
import { fetchLinksIfNeeded } from '../../state/links/actions';
import { fetchSiteMetadataIfNeeded } from '../../state/metadata/actions';
import { fetchPageIfNeeded } from '../../state/pages/actions';
import PostPreview from '../../components/PostPreview';
import SocialLinks from '../../components/SocialLinks';

const styles = theme => ({
  divider: {
    border: 0,
    borderTop: `1px solid ${theme.palette.text.secondary}`,
    height: 0,
    marginBottom: theme.spacing.unit * 6,
    maxWidth: 700,
    opacity: 0.2,
    width: '100%',
  },
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 6,
    paddingBottom: theme.spacing.unit * 6,
  },
  logo: {
    display: 'block',
    margin: '0 auto',
    maxinWidth: 700,
    overflow: 'hidden',
    paddingTop: theme.spacing.unit * 6,
    paddingBottom: theme.spacing.unit * 6,
  },
  postPreviews: {
    listStyle: 'none',
    marginBottom: theme.spacing.unit * 4,
    marginTop: theme.spacing.unit * 4,
    padding: 0,
  },
  postPreviewsHeaderContainer: {
    marginTop: theme.spacing.unit * 4,
    padding: 0,
  },
  postPreviewsHeader: {
    margin: '0 auto',
    maxWidth: 700,
    position: 'relative',
  },
  postPreviewsHeaderTitle: {
    bottom: -1,
    borderBottom: `1px solid ${theme.palette.text.secondary}`,
    display: 'inline-block',
    padding: theme.spacing.unit,
    position: 'absolute',
    left: 0,
  },
  postPreviewsHeaderContainerDivider: {
    border: 0,
    borderTop: `1px solid ${theme.palette.text.secondary}`,
    disaplay: 'block',
    height: 0,
    margin: '0 auto',
    maxWidth: 700,
    opacity: 0.2,
    padding: 0,
    width: '100%',
  },
  tag: {
    display: 'block',
    textAlign: 'center',
  },
  title: {
    display: 'block',
    textAlign: 'center',
  },
});

const Index = ({
  classes,
  metadata,
  page,
  pages,
  posts,
}) => {
  let postPreviews;
  try {
    postPreviews = pages[page].slugs.map(slug => posts[slug]);
  } catch (e) {
    postPreviews = undefined;
  }

  return (
    <div className={classes.root}>
      <Head>
        <title>{(metadata && metadata.title) || 'Cosmic Blog'}</title>
        {
          metadata && metadata.logo && metadata.logo.imgix_url &&
          <meta property="og:image" content={`${metadata.logo.imgix_url}?h=279&w=279&fit=crop&fm=png`} key="og_image" />
        }
        {
          metadata && metadata.logo && metadata.logo.imgix_url &&
          <meta property="og:image:type" content="image/png" key="og_image_type" />
        }
        {
          metadata && metadata.logo && metadata.logo.imgix_url &&
          <meta property="og:image:width" content="279" key="og_image_width" />
        }
        {
          metadata && metadata.logo && metadata.logo.imgix_url &&
          <meta property="og:image:height" content="279" key="og_image_height" />
        }
        <meta property="og:description" content="Clean, minimalist, content-first Blog powered by Cosmic JS" key="og_description" />
        <meta property="og:title" content="Cosmic Blog" key="og_title" />
        <meta property="og:url" content="http://cosmicblog.chriso.io" key="og_url" />
      </Head>
      {
        metadata && metadata.title &&
        <Typography className={classes.title} variant="display4" gutterBottom>
          {metadata.title}
        </Typography>
      }
      {
        metadata && metadata.tag &&
        <Typography className={classes.tag} variant="subheading" gutterBottom>
          {metadata.tag}
        </Typography>
      }
      {
        metadata && metadata.logo && metadata.logo.imgix_url &&
        <img
          alt=""
          className={classes.logo}
          src={`${metadata.logo.imgix_url}?w=auto&h=250&fit=crop`}
        />
      }
      <div className={classes.postPreviewsHeaderContainer}>
        <div className={classes.postPreviewsHeader}>
          <Typography className={classes.postPreviewsHeaderTitle} variant="title">
            Most Recent
          </Typography>
        </div>
        <hr className={classes.postPreviewsHeaderContainerDivider} />
      </div>
      {
        postPreviews &&
        <ul className={classes.postPreviews}>
          {
            postPreviews.map(({ post }) => <PostPreview key={post.slug} {...post} />)
          }
        </ul>
      }
      <hr className={classes.divider} />
      <SocialLinks />
    </div>
  );
};

Index.getInitialProps = async ({ query, reduxStore }) => {
  const { p } = query;
  const page = p && parseInt(p, 10) > 1 ? parseInt(p, 10) : 1;

  try {
    const { dispatch } = reduxStore;
    await Promise.all([
      dispatch(fetchLinksIfNeeded()),
      dispatch(fetchPageIfNeeded(page)),
      dispatch(fetchSiteMetadataIfNeeded()),
    ]);
  } catch (e) {
    console.error(e); // eslint-disable-line no-console
    // Do nothing
  }

  return { page };
};

Index.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  metadata: PropTypes.shape({
    logo: PropTypes.shape({
      imgix_url: PropTypes.string,
    }),
    tag: PropTypes.string,
    title: PropTypes.string,
  }).isRequired,
  page: PropTypes.number.isRequired,
  pages: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  posts: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

const mapStateToProps = state => ({
  metadata: state.metadata,
  pages: state.pages,
  posts: state.posts,
});

export const DisconnectedIndex = withStyles(styles)(Index);

export default compose(
  connect(mapStateToProps),
  withRoot,
  withStyles(styles),
)(Index);
