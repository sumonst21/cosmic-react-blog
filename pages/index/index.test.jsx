import metadata from '../../__mocks__/metadata';
import pages from '../../__mocks__/pages';
import posts from '../../__mocks__/posts';
import { DisconnectedIndex } from '.';

describe('Disconnected Index Page', () => {
  it('should render', () => {
    const wrapper = shallow((
      <DisconnectedIndex
        metadata={metadata}
        page={1}
        pages={pages}
        posts={posts}
      />
    ));
    return expect(toJson(wrapper)).toMatchSnapshot();
  });
});
