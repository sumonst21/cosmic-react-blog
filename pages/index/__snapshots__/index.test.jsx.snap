// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disconnected Index Page should render 1`] = `
<Index
  classes={
    Object {
      "divider": "Index-divider-1",
      "logo": "Index-logo-3",
      "postPreviews": "Index-postPreviews-4",
      "postPreviewsHeader": "Index-postPreviewsHeader-6",
      "postPreviewsHeaderContainer": "Index-postPreviewsHeaderContainer-5",
      "postPreviewsHeaderContainerDivider": "Index-postPreviewsHeaderContainerDivider-8",
      "postPreviewsHeaderTitle": "Index-postPreviewsHeaderTitle-7",
      "root": "Index-root-2",
      "tag": "Index-tag-9",
      "title": "Index-title-10",
    }
  }
  metadata={
    Object {
      "finishedLoadAt": 1526859679840,
      "isLoading": false,
      "logo": Object {
        "imgix_url": "ingix_url.png",
        "url": "url.png",
      },
      "startedLoadAt": 1526859679433,
      "tag": "\\"Content is king\\" -- Bill Gates",
      "title": "My Amazing Blog",
    }
  }
  page={1}
  pages={
    Object {
      "1": Object {
        "finishedLoadAt": 1526859679918,
        "isLoading": false,
        "slugs": Array [
          "a-wonderful-blog-post-about-earth",
          "another-wonderful-blog-post-about-earth",
        ],
        "startedLoadAt": 1526859679433,
      },
    }
  }
  posts={
    Object {
      "a-wonderful-blog-post-about-earth": Object {
        "post": Object {
          "_id": "5afdbf6c44d74b4e924f1d0b",
          "bucket": "5afdbcd115bef94f7421eb0d",
          "content": "<p>When I orbited the Earth in a spaceship, I saw for the first time how beautiful our planet is. Mankind, let us preserve and increase this beauty, and not destroy it!</p><p>Space, the final frontier. These are the voyages of the Starship Enterprise. Its five-year mission: to explore strange new worlds, to seek out new life and new civilizations, to boldly go where no man has gone before.</p><p>If you could see the earth illuminated when you were in a place as dark as night, it would look to you more splendid than the moon.</p><p>To be the first to enter the cosmos, to engage, single-handed, in an unprecedented duel with nature&mdash;could one dream of anything more?</p><p>We choose to go to the moon in this decade and do the other things, not because they are easy, but because they are hard, because that goal will serve to organize and measure the best of our energies and skills, because that challenge is one that we are willing to accept, one we are unwilling to postpone, and one which we intend to win.</p><p>NASA is not about the &lsquo;Adventure of Human Space Exploration&rsquo;&hellip;We won&rsquo;t be doing it just to get out there in space &ndash; we&rsquo;ll be doing it because the things we learn out there will be making life better for a lot of people who won&rsquo;t be able to go.</p><p>Problems look mighty small from 150 miles up.</p><p>That&#39;s one small step for [a] man, one giant leap for mankind.</p><p>Where ignorance lurks, so too do the frontiers of discovery and imagination.</p><p>Dinosaurs are extinct today because they lacked opposable thumbs and the brainpower to build a space program.</p>",
          "created": "2017-10-12T13:27:49.665Z",
          "created_at": "2017-10-12T13:27:49.665Z",
          "metadata": Object {
            "author": Object {
              "_id": "5afdbf6c44d74b4e924f1d09",
              "bucket": "5afdbcd115bef94f7421eb0d",
              "content": "<p>Something about John...</p>",
              "created": "2017-10-12T13:27:49.665Z",
              "created_at": "2017-10-12T13:27:49.665Z",
              "metadata": Object {
                "image": Object {
                  "imgix_url": "https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                  "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                },
              },
              "metafields": Array [
                Object {
                  "children": null,
                  "imgix_url": "https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                  "key": "image",
                  "title": "Image",
                  "type": "file",
                  "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                  "value": "9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                },
              ],
              "modified_at": "2018-05-17T17:44:12.977Z",
              "published_at": "2018-05-17T17:44:12.977Z",
              "slug": "john-doe",
              "status": "published",
              "title": "John Doe",
              "type_slug": "authors",
            },
            "hero": Object {
              "imgix_url": "https://cosmic-s3.imgix.net/56363630-af74-11e7-b864-313f959a776e-react-blog.jpg",
              "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/56363630-af74-11e7-b864-313f959a776e-react-blog.jpg",
            },
            "teaser": "<p>I don&#39;t know what you could say about a day in which you have seen four beautiful sunsets.</p><p>It suddenly struck me that that tiny pea, pretty and blue, was the Earth. I put up my thumb and shut one eye, and my thumb blotted out the planet Earth. I didn&#39;t feel like a giant. I felt very, very small....</p>",
          },
          "metafields": Array [
            Object {
              "children": null,
              "imgix_url": "https://cosmic-s3.imgix.net/56363630-af74-11e7-b864-313f959a776e-react-blog.jpg",
              "key": "hero",
              "title": "Hero",
              "type": "file",
              "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/56363630-af74-11e7-b864-313f959a776e-react-blog.jpg",
              "value": "56363630-af74-11e7-b864-313f959a776e-react-blog.jpg",
            },
            Object {
              "children": null,
              "key": "teaser",
              "title": "Teaser",
              "type": "html-textarea",
              "value": "<p>I don&#39;t know what you could say about a day in which you have seen four beautiful sunsets.</p><p>It suddenly struck me that that tiny pea, pretty and blue, was the Earth. I put up my thumb and shut one eye, and my thumb blotted out the planet Earth. I didn&#39;t feel like a giant. I felt very, very small....</p>",
            },
            Object {
              "children": null,
              "key": "author",
              "object": Object {
                "_id": "5afdbf6c44d74b4e924f1d09",
                "bucket": "5afdbcd115bef94f7421eb0d",
                "content": "<p>Something about John...</p>",
                "created": "2017-10-12T13:27:49.665Z",
                "created_at": "2017-10-12T13:27:49.665Z",
                "metadata": Object {
                  "image": Object {
                    "imgix_url": "https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                    "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                  },
                },
                "metafields": Array [
                  Object {
                    "children": null,
                    "imgix_url": "https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                    "key": "image",
                    "title": "Image",
                    "type": "file",
                    "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                    "value": "9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
                  },
                ],
                "modified_at": "2018-05-17T17:44:12.977Z",
                "published_at": "2018-05-17T17:44:12.977Z",
                "slug": "john-doe",
                "status": "published",
                "title": "John Doe",
                "type_slug": "authors",
              },
              "object_type": "authors",
              "title": "Author",
              "type": "object",
              "value": "5afdbf6c44d74b4e924f1d09",
            },
          ],
          "modified_at": "2017-10-12T17:39:54.476Z",
          "published_at": "2018-05-17T17:44:12.979Z",
          "slug": "a-wonderful-blog-post-about-earth",
          "status": "published",
          "title": "A Wonderful Blog Post About Earth",
          "type_slug": "posts",
        },
      },
      "another-wonderful-blog-post-about-earth": Object {
        "post": Object {
          "_id": "5afdbf6c44d74b4e924f1d0d",
          "bucket": "5afdbcd115bef94f7421eb0d",
          "content": "<p>That&#39;s one small step for [a] man, one giant leap for mankind.</p><p>To be the first to enter the cosmos, to engage, single-handed, in an unprecedented duel with nature&mdash;could one dream of anything more?</p><p>The dreams of yesterday are the hopes of today and the reality of tomorrow.</p><p>The sky is the limit only for those who aren&#39;t afraid to fly!</p><p>Problems look mighty small from 150 miles up.</p><p>There can be no thought of finishing for &lsquo;aiming for the stars.&rsquo; Both figuratively and literally, it is a task to occupy the generations. And no matter how much progress one makes, there is always the thrill of just beginning.</p><p>There can be no thought of finishing for &lsquo;aiming for the stars.&rsquo; Both figuratively and literally, it is a task to occupy the generations. And no matter how much progress one makes, there is always the thrill of just beginning.</p><p>There can be no thought of finishing for &lsquo;aiming for the stars.&rsquo; Both figuratively and literally, it is a task to occupy the generations. And no matter how much progress one makes, there is always the thrill of just beginning.</p><p>Houston, Tranquillity Base here. The Eagle has landed.</p><p>You know, being a test pilot isn&#39;t always the healthiest business in the world.</p>",
          "created": "2017-10-12T13:27:49.666Z",
          "created_at": "2017-10-12T13:27:49.666Z",
          "metadata": Object {
            "author": Object {
              "_id": "5afdbf6c44d74b4e924f1d06",
              "bucket": "5afdbcd115bef94f7421eb0d",
              "content": "<p>Something about Jane...</p>",
              "created": "2017-10-12T13:27:49.663Z",
              "created_at": "2017-10-12T13:27:49.663Z",
              "metadata": Object {
                "image": Object {
                  "imgix_url": "https://cosmic-s3.imgix.net/99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                  "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                },
              },
              "metafields": Array [
                Object {
                  "children": null,
                  "imgix_url": "https://cosmic-s3.imgix.net/99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                  "key": "image",
                  "title": "Image",
                  "type": "file",
                  "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                  "value": "99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                },
              ],
              "modified_at": "2018-05-17T17:44:12.973Z",
              "published_at": "2018-05-17T17:44:12.973Z",
              "slug": "jane-doe",
              "status": "published",
              "title": "Jane Doe",
              "type_slug": "authors",
            },
            "categories": Array [
              null,
              null,
            ],
            "hero": Object {
              "imgix_url": "https://cosmic-s3.imgix.net/99fd6650-23f5-11e7-875c-3f5dc9c15c2b-beach.jpg",
              "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/99fd6650-23f5-11e7-875c-3f5dc9c15c2b-beach.jpg",
            },
            "teaser": "<p>Science cuts two ways, of course; its products can be used for both good and evil. But there&#39;s no turning back from science. The early warnings about technological dangers also come from science.</p><p>Houston, Tranquillity Base here. The Eagle has landed...</p>",
          },
          "metafields": Array [
            Object {
              "children": null,
              "imgix_url": "https://cosmic-s3.imgix.net/99fd6650-23f5-11e7-875c-3f5dc9c15c2b-beach.jpg",
              "key": "hero",
              "title": "Hero",
              "type": "file",
              "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/99fd6650-23f5-11e7-875c-3f5dc9c15c2b-beach.jpg",
              "value": "99fd6650-23f5-11e7-875c-3f5dc9c15c2b-beach.jpg",
            },
            Object {
              "children": null,
              "key": "teaser",
              "title": "Teaser",
              "type": "html-textarea",
              "value": "<p>Science cuts two ways, of course; its products can be used for both good and evil. But there&#39;s no turning back from science. The early warnings about technological dangers also come from science.</p><p>Houston, Tranquillity Base here. The Eagle has landed...</p>",
            },
            Object {
              "children": null,
              "key": "author",
              "object": Object {
                "_id": "5afdbf6c44d74b4e924f1d06",
                "bucket": "5afdbcd115bef94f7421eb0d",
                "content": "<p>Something about Jane...</p>",
                "created": "2017-10-12T13:27:49.663Z",
                "created_at": "2017-10-12T13:27:49.663Z",
                "metadata": Object {
                  "image": Object {
                    "imgix_url": "https://cosmic-s3.imgix.net/99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                    "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                  },
                },
                "metafields": Array [
                  Object {
                    "children": null,
                    "imgix_url": "https://cosmic-s3.imgix.net/99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                    "key": "image",
                    "title": "Image",
                    "type": "file",
                    "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                    "value": "99f48cb0-23f5-11e7-875c-3f5dc9c15c2b-female.jpg",
                  },
                ],
                "modified_at": "2018-05-17T17:44:12.973Z",
                "published_at": "2018-05-17T17:44:12.973Z",
                "slug": "jane-doe",
                "status": "published",
                "title": "Jane Doe",
                "type_slug": "authors",
              },
              "object_type": "authors",
              "title": "Author",
              "type": "object",
              "value": "5afdbf6c44d74b4e924f1d06",
            },
            Object {
              "children": null,
              "key": "categories",
              "object_type": "categories",
              "objects": Array [
                null,
                null,
              ],
              "title": "Categories",
              "type": "objects",
              "value": "58f565290687f7353b0009bb,58f565210687f7353b0009ba",
            },
          ],
          "modified_at": "2018-05-17T17:44:12.981Z",
          "published_at": "2018-05-17T17:44:12.981Z",
          "slug": "another-wonderful-blog-post-about-earth",
          "status": "published",
          "title": "Another Wonderful Blog Post About Earth",
          "type_slug": "posts",
        },
      },
    }
  }
/>
`;
